src/routes/tools/local-climate-change-snapshot/_data.js

```diff
@@ -200,3 +221,60 @@ export function getQueryParams({ location, boundary, imperial = true, stat }) {
   }
   return { params, method };
 }
+
+/**
+ * Calculate 30 year average for each scenario & period combination
+ * @param {array} scenarios
+ * @param {array} periods
+ * @return {array} scenario & period combinations that have a valid 30 year avg value
+ */
+export function calc30yAvgByPeriod(scenarios, periods = DEFAULT_STAT_PERIODS) {
+  const data = scenarios.map((d) => {
+    const { values, id: scenarioId, label: scenarioLabel } = d;
+    const dataByPeriods = periods.map((period) => {
+      const { id: periodId, label: periodLabel, start, end } = period;
+      const filteredValues = values.filter(
+        ({ date }) =>
+          date.getUTCFullYear() >= start && date.getUTCFullYear() <= end
+      );
+      let avg = null;
+      if (filteredValues.length) {
+        avg = mean(filteredValues, (d) => d.value);
+      }
+      return { periodId, periodLabel, scenarioId, scenarioLabel, avg };
+    });
+    return [...dataByPeriods];
+  });
+  return merge(data).filter(({ avg }) => avg);
```

Comment:
if `avg` is ever `0` here then those values will be filtered out, is that intended?